pkg/metrics/metrics.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	v1 "k8s.io/api/core/v1"
+)
+
+// ExposeMetricsPort generate a Kubernetes Service to expose metrics port
+func ExposeMetricsPort() (*v1.Service, error) {
```

Comment:
This is such a nit, but what if we change this to `ExposePort()`
The call changes from `metrics.ExposeMetricsPort` to `metrics.ExposePort` which I think is just as clear and has less redundancy.

Reply:
Yes, I wasn't happy with that naming, not sure `ExposePort ` is the way to go either. This function creates service, exposes port and starts to listen and serve for prom. registry. So maybe `metrics.SetupPrometheus()` would make more sense? WDYT?

Reply:
What about `metrics.Start()` or `metrics.Expose() or `metrics.Setup()`.

Reply:
`metrics.Setup()` sounds good to me, will do. Only wanted to include Prometheus in the name to make it more clear it's specific to it.